modules/claims_api/app/controllers/claims_api/v2/application_controller.rb

```diff
@@ -115,6 +119,40 @@ def user_represents_veteran?
 
         rep.poa_codes.include?(veteran_poa_code)
       end
+
+      private
+
+      def validate_ccg_token!
+        client = TokenValidation::V2::Client.new(api_key: Settings.claims_api.token_validation.api_key)
+        root_url = request.base_url == 'http://localhost:3000' ? 'https://sandbox-api.va.gov' : request.base_url
+        claims_audience = "#{root_url}/services/claims"
+        request_method_to_scope = {
+          'GET' => 'claim.read',
+          'PUT' => 'claim.write',
+          'POST' => 'claim.write'
+        }
+
+        @is_valid_ccg_flow ||= client.token_valid?(audience: claims_audience,
+                                                   scope: request_method_to_scope[request.method],
+                                                   token: token)
+        raise ::Common::Exceptions::Forbidden unless @is_valid_ccg_flow
+      end
```

Comment:
seems like overkill to abstract/consolidate with a single duplication, but just noting that this is a duplication of the same `validate_ccg_token` method in the v1 `ApplicationController`